src/posit/connect/content.py

```diff
@@ -183,6 +183,94 @@ def app_role(self) -> str:
     def id(self) -> str:
         return self.get("id")  # type: ignore
 
+    def update(  # type: ignore
+        self,
+        name: Optional[str] = None,
+        title: Optional[str] = None,
+        description: Optional[str] = None,
+        access_type: Optional[str] = None,
+        owner_guid: Optional[str] = None,
+        connection_timeout: Optional[int] = None,
+        read_timeout: Optional[int] = None,
+        init_timeout: Optional[int] = None,
+        idle_timeout: Optional[int] = None,
+        max_processes: Optional[int] = None,
+        min_processes: Optional[int] = None,
+        max_conns_per_process: Optional[int] = None,
+        load_factor: Optional[float] = None,
+        cpu_request: Optional[float] = None,
+        cpu_limit: Optional[float] = None,
+        memory_request: Optional[int] = None,
+        memory_limit: Optional[int] = None,
+        amd_gpu_limit: Optional[int] = None,
+        nvidia_gpu_limit: Optional[int] = None,
+        run_as: Optional[str] = None,
+        run_as_current_user: Optional[bool] = None,
+        default_image_name: Optional[str] = None,
+        default_r_environment_management: Optional[bool] = None,
+        default_py_environment_management: Optional[bool] = None,
+        service_account_name: Optional[str] = None,
```

Comment:
For forward compatibility, would it make sense to accept `**kw` here in addition to the known/typed attributes? The behavior could be to issue a warning (in case it's a simple typo) and merge the values into the payload. If this makes the payload invalid, Connect will return a 400.

Reply:
We could do that. You still get the benefit of tab completion and type hinting for the known parameters, and like you say, if the thing you send is invalid, the server will reject it.

Reply:
Added support for this in the latest iteration.